x-pack/plugins/security_solution/cypress/e2e/detection_rules/prebuilt_rules_install_update_workflows.cy.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  GO_BACK_TO_RULES_TABLE_BUTTON,
+  INSTALL_ALL_RULES_BUTTON,
+  INSTALL_SELECTED_RULES_BUTTON,
+  RULES_MANAGEMENT_TABLE,
+  RULES_ROW,
+  RULES_UPDATES_TABLE,
+  SELECT_ALL_RULES_ON_PAGE_CHECKBOX,
+  TOASTER,
+  UPGRADE_ALL_RULES_BUTTON,
+} from '../../screens/alerts_detection_rules';
+import { waitForRulesTableToBeLoaded } from '../../tasks/alerts_detection_rules';
+import {
+  createNewRuleAsset,
+  getRuleAssets,
+  installAllPrebuiltRulesRequest,
+  preventPrebuiltRulesPackageInstallation,
+} from '../../tasks/api_calls/prebuilt_rules';
+import { resetRulesTableState, deleteAlertsAndRules, reload } from '../../tasks/common';
+import { esArchiverLoad, esArchiverResetKibana } from '../../tasks/es_archiver';
+import { login, visitWithoutDateRange } from '../../tasks/login';
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+import { addElasticRulessButtonClick, ruleUpdatesTabClick } from '../../tasks/prebuilt_rules';
+
+describe('Detection rules, Prebuilt Rules Installation and Update workflow', () => {
+  beforeEach(() => {
+    login();
+    resetRulesTableState();
+    deleteAlertsAndRules();
+    esArchiverResetKibana();
+
+    visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+  });
+
+  after(() => {
+    // Reload after using esArchiverResetKibana to
+    // prevent flakyness in unrelated test suites
+    esArchiverLoad('auditbeat');
+    esArchiverLoad('network');
+  });
+
+  describe('Installation of prebuilt rules package via Fleet', () => {
+    beforeEach(() => {
+      cy.intercept('POST', '/api/fleet/epm/packages/_bulk*').as('installPackage');
+      waitForRulesTableToBeLoaded();
+    });
+
+    it('should install package from Fleet in the background', () => {
+      /* Assert that the package in installed from Fleet by checking that
+      /* the installSource is "registry", as opposed to "bundle" */
+      cy.wait('@installPackage', {
+        timeout: 60000,
+      }).then(({ response }) => {
+        cy.wrap(response?.statusCode).should('eql', 200);
+
+        const packages = response?.body.items.map(({ name, result }: BulkInstallPackageInfo) => ({
+          name,
+          installSource: result.installSource,
+        }));
+
+        expect(packages.length).to.have.greaterThan(0);
+        expect(packages).to.deep.include.members([
+          { name: 'security_detection_engine', installSource: 'registry' },
+        ]);
+      });
+    });
+
+    it('should install rules from the Fleet package when user clicks on CTA', () => {
+      /* Retrieve how many rules were installed from the Fleet package */
+      cy.wait('@installPackage', {
+        timeout: 60000,
+      }).then(() => {
+        getRuleAssets().then((response) => {
+          const ruleIds = response.body.hits.hits.map(
+            (hit: { _source: { ['security-rule']: Rule } }) => hit._source['security-rule'].rule_id
+          );
+
+          const numberOfRulesToInstall = [...new Set(ruleIds)].length;
```

Comment:
Because the rules package can (actually, will) contain historical versions of the same rules, which have the same `rule_id`. In the UI we only take the latest historical version of each rule, but for this test we need to filter those out.

I will apply your suggestion though, thanks!